nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -822,12 +830,16 @@ in
               wantedBy = [ "machines.target" ];
               wants = [ "network.target" ];
               after = [ "network.target" ];
-              restartTriggers = [
-                containerConfig.path
-                config.environment.etc."${configurationDirectoryName}/${name}.conf".source
-              ];
-              restartIfChanged = true;
-            }
+            } // (
+            optionalAttrs containerConfig.restartIfChanged
+              {
+                restartTriggers = [
+                  containerConfig.path
+                  config.environment.etc."${configurationDirectoryName}/${name}.conf".source
+                ];
+                restartIfChanged = true;
```

Comment:
I think setting `restartIfChanged = containerConfig.restartIfChanged` is enough.  There is no need to change `restartTriggers`.

Reply:
Just tested and the simplification works, thanks